### PR TITLE
Add support to amigaos

### PR DESCRIPTION
**Pull request by Wolf3s**

### CODE REVIEW

**Comment by Wolf3s** [comment]
@sahlberg i removed the password patch and i dedcided to put in my test branch

**Comment by Wolf3s** [comment]
Draft until i solve the getpid portability.

File: lib/init.c [hunk 1]
```diff
@@ -84,6 +84,19 @@
...
+#ifndef __AROS__
+#define srandom srand
+#define random rand
+#endif
+#ifndef __amigaos4__
+#include <proto/bsdsocket.h>
+#define close CloseSocket
+#endif
+#endif // __amigaos4__
+
```
Comment (sahlberg) [inline lib/init.c:99]: Lets move this into compat.h

File: lib/init.c [hunk 2]
```diff
@@ -192,8 +205,11 @@ struct smb2_url *smb2_parse_url(struct smb2_context *smb2, const char *url)
                 smb2_set_error(smb2, "URL is too long");
                 return NULL;
         }
+#if defined(__amigaos4__) || defined(__AMIGA__) || defined(__AROS__)
+        strcpy(str, url + 6);
+#else
         strncpy(str, url + 6, MAX_URL_SIZE);
-
+#endif
```
Comment (sahlberg) [inline lib/init.c:212]: Add a 
#if defined(__amigaos4__) || defined(__AMIGA__) || defined(__AROS__)
#define strncpy(a,b,c) strcpy(a,b)
#endif

to compat.h

File: lib/libsmb2.c [hunk 3]
```diff
@@ -84,10 +84,18 @@
...
 #include "asprintf.h"
 #endif
 
+#if !defined(__amigaos4__) && (defined(__AMIGA__) || defined(__AROS__))
+#include <proto/bsdsocket.h>
+#undef getaddrinfo
+#undef freeaddrinfo
+#define close CloseSocket
+#endif
+
```
Comment (sahlberg) [inline lib/libsmb2.c:97]: move to compat.h ?

File: lib/socket.c [hunk 4]
```diff
@@ -28,6 +28,16 @@
...
 
+#if !defined(__amigaos4__) && (defined(__AMIGA__) || defined(__AROS__))
+#include <sys/ioctl.h>
+#include <proto/bsdsocket.h>
+#undef getaddrinfo
+#undef freeaddrinfo
+#undef HAVE_UNISTD_H
+#define close CloseSocket
+#endif
+
```
Comment (sahlberg) [inline lib/socket.c:39]: moce to compat.h ?

File: examples/smb2-cat-async.c [hunk 5]
```diff
@@ -36,7 +48,11 @@ int usage(void)
         fprintf(stderr, "Usage:\n"
                 "smb2-cat-async <smb2-url>\n\n"
                 "URL format: "
+#ifdef USE_PASSWORD
```
[the remark this answers is not in the document]
Comment (Wolf3s) [reply examples/smb2-cat-async.c]: This USE_PASSWORD macro i removed 3 weeks ago:
see here: https://github.com/sahlberg/libsmb2/commit/0f63e7d7471898d79750efd10a37238e9a744ea0

File: lib/libsmb2.c (hunk 3 above)
Comment (Wolf3s) [reply lib/libsmb2.c]: moved here: https://github.com/sahlberg/libsmb2/pull/297/commits/99944b39e6ceebf7603c3dd189222a666a8643aa

File: lib/socket.c (hunk 4 above)
Comment (Wolf3s) [reply lib/socket.c]: Moved here: https://github.com/sahlberg/libsmb2/pull/297/commits/99944b39e6ceebf7603c3dd189222a666a8643aa

File: lib/init.c (hunk 1 above)
Comment (Wolf3s) [reply lib/init.c]: Moved here:
https://github.com/sahlberg/libsmb2/pull/297/commits/99944b39e6ceebf7603c3dd189222a666a8643aa

File: lib/init.c (hunk 2 above)
Comment (Wolf3s) [reply lib/init.c]: Added here:
https://github.com/sahlberg/libsmb2/pull/297/commits/99944b39e6ceebf7603c3dd189222a666a8643aa

**Pull request merged**